tests/e2e/vsan_stretched_cluster_utils.go

```diff
@@ -913,3 +917,74 @@ func createStaticPvAndPvcInParallel(client clientset.Interface, ctx context.Cont
 
 	}
 }
+
+// triggerFullSyncInParallel triggers full sync on demand in parallel
+func triggerFullSyncInParallel(ctx context.Context, client clientset.Interface, wg *sync.WaitGroup) {
+	defer wg.Done()
+	restConfig := getRestConfigClient()
+	crd := getTriggerFullSyncCrd(ctx, client, restConfig)
+	updateTriggerFullSyncCrd(ctx, restConfig, crd)
+	err := waitForFullSyncToFinish(client, ctx, restConfig)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Full sync did not finish in given time")
+	crd = getTriggerFullSyncCrd(ctx, client, restConfig)
+	updateTriggerFullSyncCrd(ctx, restConfig, crd)
+}
+
+// getTriggerFullSyncCrd fetches full sync crd from the list of crds in k8s cluster
+func getTriggerFullSyncCrd(ctx context.Context, client clientset.Interface,
+	restConfig *rest.Config) triggercsifullsyncv1alpha1.TriggerCsiFullSync {
+	cnsOperatorClient, err := k8s.NewClientForGroup(ctx, restConfig, cnsoperatorv1alpha1.GroupName)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	crds := &triggercsifullsyncv1alpha1.TriggerCsiFullSyncList{}
+	err = cnsOperatorClient.List(ctx, crds)
+	framework.Logf("Error is %v, list of crds is %v", err, crds)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	crdFound := false
+
+	var fullSyncCrd triggercsifullsyncv1alpha1.TriggerCsiFullSync
+	for _, crd := range crds.Items {
+		framework.Logf("crd name is %v", crd.Name)
+		if crd.Name == crdtriggercsifullsyncsName {
+			framework.Logf("Found TriggerFullSync crd: %v", crd.Name)
+			fullSyncCrd = crd
+			crdFound = true
+			break
+		}
+
+	}
+	gomega.Expect(crdFound).To(gomega.BeTrue(), "couldn't find full sync crd: %s", crdtriggercsifullsyncsName)
+	return fullSyncCrd
+}
+
+// updateTriggerFullSyncCrd triggers full sync by updating TriggerSyncID
+// value to  LastTriggerSyncID +1 in full sync crd
+func updateTriggerFullSyncCrd(ctx context.Context, restConfig *rest.Config,
+	crd triggercsifullsyncv1alpha1.TriggerCsiFullSync) {
+	cnsOperatorClient, err := k8s.NewClientForGroup(ctx, restConfig, cnsoperatorv1alpha1.GroupName)
```

Comment:
I don't like how we are creating this client in each subroutine, can you create this in the testcase and pass that to these methods

Reply:
Done